ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py

```diff
@@ -3058,3 +3058,87 @@ def test_tensorflow_Elu(
         features=x[0],
         name=name,
     )
+
+
+@st.composite
+def _conv2d_helper(draw):
+    input_dtype, \
+        x, \
+        filters, \
+        strides, \
+        dilations, \
+        data_format, \
+        padding = draw(_x_and_filters(
+        dtypes=helpers.get_dtypes("float", full=False),
+        data_format=st.sampled_from(["NHWC", "NCHW"]),
+        padding=st.sampled_from(["SAME", "VALID", "EXPLICIT"]),
+        type="2d",
+        dilation_min=1,
+        dilation_max=1,
+    ))
+    if padding == "EXPLICIT":
+        pad_top = draw(st.integers(min_value=0, max_value=3))
+        pad_bottom = draw(st.integers(min_value=0, max_value=3))
+        pad_left = draw(st.integers(min_value=0, max_value=3))
+        pad_right = draw(st.integers(min_value=0, max_value=3))
+        explicit_paddings = [(pad_top, pad_bottom), (pad_left, pad_right)]
+        if data_format == "NHWC":
+            explicit_paddings = [(0, 0), *explicit_paddings, (0, 0)]
+        else:
+            explicit_paddings = [(0, 0), (0, 0), *explicit_paddings]
+    else:
+        explicit_paddings = []
+
+    return input_dtype, \
+        x, \
+        filters, \
+        dilations, \
+        data_format, \
+        strides, \
+        padding, \
+        explicit_paddings
+
+
+@handle_frontend_test(
+    fn_tree="tensorflow.raw_ops.Conv2D",
+    x_f_d_df=_conv2d_helper(),
+    test_with_out=st.just(False),
+)
+def test_tensorflow_Conv2D(
+    *,
+    x_f_d_df,
+    test_flags,
+    frontend,
+    fn_tree,
+    on_device,
+):
+    input_dtype, \
+        x, \
+        filters, \
+        dilation, \
+        data_format, \
+        stride, \
+        paddings,\
+        explicit_paddings = x_f_d_df
+    # Broadcast strides and dilations to correct dims for the ground truth
+    # backend func to run correctly
+    stride = _convolution_broadcast_helper(
+        stride, num_spatial_dims=2, channel_index=3, name="strides"
+    )
+    dilation = _convolution_broadcast_helper(
+        dilation, num_spatial_dims=2, channel_index=3, name="dilations"
+    )
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        test_flags=test_flags,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=x,
+        filter=filters,
+        strides=stride,
+        padding=paddings,
+        explicit_paddings=explicit_paddings,
```

Comment:
try adding `use_cudnn_on_gpu=True` between `padding` and `explicit_paddings`

Reply:
ok will do that now

Reply:
running the tests again, let's see

Reply:
@AnnaTz  still shows the same error, for numpy 

```
def _dilate_pad_conv(x, filters, strides, padding, dims, dilations):
        for j in range(dims):
>           if dilations[j] > 1:
E           TypeError: '>' not supported between instances of 'str' and 'int'
E           Falsifying example: test_tensorflow_Conv2D(
```
and for tensorflow as well 

```
ivy/functional/backends/tensorflow/layers.py:25: in _pad_before_conv
    filter_shape = [
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <range_iterator object at 0x7f11485020c0>

    filter_shape = [
>       filter_shape[i] + (filter_shape[i] - 1) * (dilations[i] - 1)
        for i in range(dims)
    ]
E   TypeError: unsupported operand type(s) for -: 'str' and 'int'
E   Falsifying example: test_tensorflow_Conv2D(
```

Reply:
the dilations is set to `Any`, can we make it  integer somehow

Reply:
I suggest you use the debugger to find out what's wrong. See what dilation is being generated by `_conv2d_helper`, and what is the string value it has when it reaches `_dilate_pad_conv`.
 I can tell you it's not a problem with `_dilate_pad_conv`. This helper is being used in many other convolution functions without a problem. The problem is probably in how the arguments are being passed.

Reply:
yeah I thought the same, the problem is not with `_dilate_pad_conv` but with the `_conv2d_helper`, i'll check using the debugger

Reply:
@AnnaTz  it shows the same stuff, `> is not supported for str and int`, 
and is the gpu important for testing the raw_ops file, i reinstalled the cuda drivers but it is not working.

Reply:
like we used` use_cudnn_on_gpu=True`, but is it very important or we can leave it

Reply:
> @AnnaTz it shows the same stuff, `> is not supported for str and int`, and is the gpu important for testing the raw_ops file, i reinstalled the cuda drivers but it is not working.

did you use breakpoints to see the value of dilations in different stages of the code?

Reply:
yeah i did, it shows that the dilation is getting the values `"NHWC" `instead of `[1,1,1,1].]`

Reply:
Then, `_x_and_filters` is somehow mixing up the dilation with data_format. Try a breaking point there.